internal/controllers/core/kubernetesdiscovery/reconciler.go

```diff
@@ -211,19 +219,25 @@ func (w *Reconciler) addOrReplace(ctx context.Context, key watcherID, kd *store.
 		extraSelectors = append(extraSelectors, selector)
 	}
 
-	if _, ok := w.watchers[key]; ok {
+	if _, ok := w.watchers[watcherKey]; ok {
 		// if a watcher already exists, just tear it down and we'll set it up from scratch so that
 		// we don't have to diff a bunch of different pieces
-		w.teardown(key)
+		w.teardown(watcherKey)
+	}
+
+	kCli, err := w.clients.GetK8sClient(types.NamespacedName{Namespace: kd.Namespace, Name: kd.Spec.Cluster})
+	if err != nil {
+		return fmt.Errorf("cluster %q connection error: %v", kd.Spec.Cluster, err)
```

Comment:
i don't think this should be treated as a reconciler error? we don't want to re-reconcile if this happens.

if anything, it feels like it should be some sort of waiting status

Reply:
You're right, this shouldn't retry since it requires the `Cluster` to change which will implicitly trigger a re-reconciliation.

Agreed on status, I'll update the `KubernetesDiscoveryStatus` to add those fields in a new PR and pull changes back in here